src/main/java/com/faforever/api/data/domain/ModerationReport.java

```diff
@@ -0,0 +1,193 @@
+import com.faforever.api.data.checks.permission.IsModerator;
+import com.faforever.api.security.FafUserDetails;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.yahoo.elide.annotation.Audit;
+import com.yahoo.elide.annotation.Audit.Action;
+import com.yahoo.elide.annotation.CreatePermission;
+import com.yahoo.elide.annotation.DeletePermission;
+import com.yahoo.elide.annotation.Include;
+import com.yahoo.elide.annotation.OnCreatePreSecurity;
+import com.yahoo.elide.annotation.OnUpdatePreSecurity;
+import com.yahoo.elide.annotation.ReadPermission;
+import com.yahoo.elide.annotation.UpdatePermission;
+import com.yahoo.elide.core.RequestScope;
+import lombok.EqualsAndHashCode;
+import lombok.Setter;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.Collection;
+import java.util.Set;
+
+@Entity
+@Table(name = "moderation_report")
+@Setter
+@EqualsAndHashCode(of = {"game", "reporter", "reportDescription"}, callSuper = false)
+@Include(rootLevel = true, type = ModerationReport.TYPE_NAME)
+@ReadPermission(expression = IsEntityOwner.EXPRESSION + " OR " + IsModerator.EXPRESSION)
+@DeletePermission(expression = Prefab.NONE)
+@CreatePermission(expression = Prefab.ALL)
+@Audit(action = Action.CREATE, logStatement = "Moderation report `{0}` has been reported", logExpressions = "${moderationReport}")
+@Audit(action = Action.UPDATE, logStatement = "Moderation report `{0}` has been updated", logExpressions = "${moderationReport}")
+public class ModerationReport extends AbstractEntity implements OwnableEntity {
+  public static final String TYPE_NAME = "moderationReport";
+  private ModerationReportStatus reportStatus;
+  private Player reporter;
+  private String reportDescription;
+  private String gameIncidentTimecode;
+  private Game game;
+  private String moderatorNotice;
+  private String moderatorPrivateNote;
+  private Player lastModerator;
+  private Set<Player> reportedUsers;
+  private Collection<BanInfo> bans;
+
+  @NotNull
+  @Column(name = "report_status")
+  @Enumerated(EnumType.STRING)
+  @CreatePermission(expression = Prefab.NONE)
+  @UpdatePermission(expression = IsModerator.EXPRESSION) // or IsInAwaitingState is hack
+  public ModerationReportStatus getReportStatus() {
+    return reportStatus;
+  }
+
+  @NotNull
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "reporter_id", referencedColumnName = "id", nullable = false)
```

Comment:
nullable = false is not used anywhere else so for code styling reasons a would leave it out right?